invokeai/app/invocations/prompt.py

```diff
@@ -55,3 +56,35 @@ def invoke(self, context: InvocationContext) -> PromptCollectionOutput:
             prompts = generator.generate(self.prompt, num_images=self.max_prompts)
 
         return PromptCollectionOutput(prompt_collection=prompts, count=len(prompts))
+    
+
+class PromptsFromFileInvocation(BaseInvocation):
+    '''Loads prompts from a text file'''
+    # fmt: off
+    type: Literal['prompt_from_file'] = 'prompt_from_file'
+
+    # Inputs
+    filename: str = Field(default=None, description="Filename of prompt text file")
+    pre_prompt: str = Field(default=None, description="Add to start of prompt")
+    post_prompt: str = Field(default=None, description="Add to end of prompt")
```

Comment:
- these two are optional, so we need to provide types to represent that
- you may omit the `default=None` when making a field optional
- I think a clearer description would be like "String to prepend to each prompt" and "String to append to each prompt"

Example of types:
```py
pre_prompt: Optional[str] = Field(description="Add to start of prompt")
```